core/src/main/scala/org/apache/spark/util/SparkUncaughtExceptionHandler.scala

```diff
@@ -48,11 +48,24 @@ private[spark] class SparkUncaughtExceptionHandler(val exitOnUncaughtException:
             System.exit(SparkExitCode.OOM)
           case _ if exitOnUncaughtException =>
             System.exit(SparkExitCode.UNCAUGHT_EXCEPTION)
+          case _ =>
+            // SPARK-30310: Don't System.exit() when exitOnUncaughtException is false
         }
       }
     } catch {
-      case oom: OutOfMemoryError => Runtime.getRuntime.halt(SparkExitCode.OOM)
-      case t: Throwable => Runtime.getRuntime.halt(SparkExitCode.UNCAUGHT_EXCEPTION_TWICE)
+      case oom: OutOfMemoryError =>
+        try { logError(s"Uncaught OutOfMemoryError in thread $thread, process halted.", oom) }
```

Comment:
Thanks for the update! It's a nit, but maybe better to be consistent on style:

```
try {
...
} catch {
...
}
```

as one-liner of try statement and multi-lines or catch statement look inconsistent.

Reply:
I actually agree!  Will modify.